### PR TITLE
Integrate Dynamic Settings

### PR DESCRIPTION
**Pull request by davidshq**
## Dynamic Settings
- Add two new models: `setting` and `settings_group` (these hold and group the dynamic settings).
- Use the built-in Python `logging` library in Project model and `tasks.py`. 
- Fields added to the Project model that map to the Settings model.
- Remove `default_identified_artifacts` and pass an empty array at the global level instead.
   - This is per conversation in PR 574 in OpenImaging MIQA repo: https://github.com/OpenImaging/miqa/pull/574 where it was requested that transforming of artifacts be handled on the client (Vue) side and be set per-project.
- Modify `tasks.py` so that it can use the dynamic settings for determining the learning models to load.
- Add a `setting.json` file which contains a base configuration for MIQA that can be imported when setting up a new instance.

## Misc
- Add an AUTO_ADVANCE flag which determines whether scans advanced automatically after a decision is rendered.
- Update GitHub Actions yaml files to use most recent versions of actions available atm.
- Disabling Playwright temporarily from CI due to need to setup backend for frontend testing.
- Update `.gitattributes` to fix some issues with some binary files being detected as changed even when they aren't.
- Exclude test files from extraneous dependencies ESLint rule.

### CODE REVIEW

**Comment by davidshq** [comment]
@annehaley - this one should be ready to go except that for some reason the CI is having trouble finding itk. not sure if you have any ideas? I reverted the github actions I'd updated but that doesn't seem to have resolved the issue

File: web_client/src/types.ts [hunk 1]
```diff
@@ -53,7 +58,9 @@ interface Frame {
   scan: string,
   extension: string,
   experiment?: string,
+  // TODO: Passed in from Python as { results: {}, evaluation_model: string }
   frame_evaluation?: string,
+  // TODO: Also has download_url
```
Comment (annehaley) [inline web_client/src/types.ts]: When serialized, the `frame_evaluation` field is passed to the webapp as the id of the `FrameEvaluation` object; it's not nested into the response. Instead, we use subsequent requests for the target `FrameEvaluation`s using the id, which is a UUID string.

File: web_client/src/types.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+/* eslint camelcase: "off" */
 /* eslint no-unused-vars: "off" */
 /* eslint no-shadow: "off" */
 import type { WorkerPool } from 'itk/WorkerPool';
 
+// TODO: Throughout we have snake case properties, these
+// could be converted to camel case using:
+// https://morioh.com/p/8e8b33c25ea1
+
```
Comment (annehaley) [inline web_client/src/types.ts]: We use snake case in the type file because that is how the keys are written when they are received from the backend, therefore request responses have snake case keys

File: miqa/core/migrations/0023_user_identified_artifacts.py [hunk 3]
```diff
@@ -14,8 +12,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='scandecision',
             name='user_identified_artifacts',
-            field=models.JSONField(
-                default=miqa.core.models.scan_decision.default_identified_artifacts
-            ),
+            field=models.JSONField(default=dict),
```
Comment (annehaley) [inline miqa/core/migrations/0023_user_identified_artifacts.py:15]: In general, it's bad practice to edit a django migration file manually. For example, when I ran migrate after these changes, this change was not applied because django does not look back in the old migration files if it already has record of applying them. I'm going to undo these changes and include them in 0036.

File: setting.json
Comment (annehaley) [inline setting.json:1]: @davidshq How can I use this file and ingest these objects to the database? The branch does not work without these objects existing, so if you tell me how to use the contents, I can include it as a post-action to the migration.

Comment (davidshq) [reply setting.json:1]: https://github.com/davidshq/miqa-notes/blob/main/quick-references/quick-reference-django.md

`./manage.py loaddata setting.json`

Comment (annehaley) [reply setting.json:1]: Thanks. I added a section to the migration file to include this as a migration step: [5a00963](https://github.com/knowledgevis/miqa-annotate/pull/15/commits/5a009632ce11ddcdd1e779dcd2858d19491b1705)

**Review by annehaley — APPROVED** [review]
